src/scvi/external/contrastivevi/_contrastive_data_splitting.py

```diff
@@ -81,15 +81,15 @@ def __init__(
                 self.n_background,
                 self.train_size,
                 self.validation_size,
-                self.data_loader_kwargs.pop("batch_size", settings.batch_size),
-                self.drop_last,
+                self.data_loader_kwargs.pop("batch_size", settings.batch_size) if
+                self.train_size_was_none and not self.drop_last else None,
             )
             self.n_target_train, self.n_target_val = validate_data_split(
                 self.n_target,
                 self.train_size,
                 self.validation_size,
-                self.data_loader_kwargs.pop("batch_size", settings.batch_size),
-                self.drop_last,
+                self.data_loader_kwargs.pop("batch_size", settings.batch_size) if
```

Comment:
Are we sure that the pop is ok and it still uses the correct batch size afterward?

Reply:
pop is what we have now in the main branch. tests are ok.